src/rest_vol_dataset.c

```diff
@@ -982,14 +997,48 @@ RV_dataset_write(size_t count, void *dset[], hid_t mem_type_id[], hid_t _mem_spa
         printf("-> %lld points selected in memory dataspace\n\n", mem_select_npoints);
 #endif
 
+        /* Handle conversion from memory datatype to file datatype, if necessary */
+        if ((needs_tconv = RV_need_tconv(transfer_info[i].file_type_id, transfer_info[i].mem_type_id)) < 0)
+            FUNC_GOTO_ERROR(H5E_DATATYPE, H5E_BADVALUE, FAIL, "unable to check if datatypes need conversion");
+
+        if (needs_tconv) {
+
+#ifdef RV_CONNECTOR_DEBUG
+            printf("-> Beginning type conversion for write\n");
+#endif
+            if ((file_type_size = H5Tget_size(transfer_info[i].file_type_id)) == 0)
+                FUNC_GOTO_ERROR(H5E_DATATYPE, H5E_BADVALUE, FAIL, "unable to get size of file datatype");
+
+            if ((mem_type_size = H5Tget_size(transfer_info[i].mem_type_id)) == 0)
+                FUNC_GOTO_ERROR(H5E_DATATYPE, H5E_BADVALUE, FAIL, "unable to get size of memory datatype");
+
+            /* Initialize type conversion */
+            RV_tconv_init(transfer_info[i].mem_type_id, &mem_type_size, transfer_info[i].file_type_id,
+                          &file_type_size, (size_t)file_select_npoints, TRUE, FALSE,
+                          &transfer_info[i].tconv_buf, &transfer_info[i].bkg_buf, NULL, &fill_bkg);
+
+            /* Perform type conversion on response values */
+            memset(transfer_info[i].tconv_buf, 0, file_type_size * (size_t)mem_select_npoints);
```

Comment:
Since TRUE is passed above for `clear_tconv_buf`, there's no need to memset it here since it will be allocated with calloc